tests/tests_pytorch/tuner/test_lr_finder.py

```diff
@@ -434,3 +435,25 @@ def test_if_lr_finder_callback_already_configured():
 
     with pytest.raises(MisconfigurationException, match="Trainer is already configured with a .* callback"):
         trainer.tune(model)
+
+
+def test_lr_finder_with_ddp(tmpdir):
+    seed_everything(7)
+
+    init_lr = 1e-4
+    dm = ClassifDataModule()
+    model = ClassificationModel(lr=init_lr)
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        strategy="ddp",
+        devices=2,
+        accelerator="cpu",
+    )
+
+    trainer.tuner.lr_find(model, datamodule=dm, update_attr=True, num_training=20)
+    lr = trainer.lightning_module.lr
+    lr = trainer.strategy.broadcast(lr)
```

Comment:
The tuner should guarantee that the learning rate suggestion (and all the picked learning rate through the scheduler) are the same on all ranks. So the test shouldn't broadcast this. Instead, it should all_gather and assert that all LR's are the same value.

Reply:
they can't be the same since it is determined from the lr vs loss curve, and even though lr values will be the same, loss can't be on each GPU process.

Reply:
Okay. And what about the learning rates at the intermediate steps? I couldn't find where we are synchronizing this.

Reply:
learning rate at intermediate steps will be the same.. only losses will be different.

Reply:
sorry, the learning rates will be the same because of the scheduler, I agree. What I meant was that the tracked losses will be different and thus also the final suggested learning rate.